src/lake/Lake/CLI/Init.lean

```diff
@@ -233,10 +233,28 @@ def initPkg (dir : FilePath) (name : String) (tmp : InitTemplate) (env : Lake.En
     else
       logWarning "failed to initialize git repository"
 
-def init (pkgName : String) (tmp : InitTemplate) (env : Lake.Env) : LogIO PUnit :=
-  initPkg "." pkgName tmp env
-
-def new (pkgName : String) (tmp : InitTemplate) (env : Lake.Env) : LogIO PUnit := do
-  let dirName := pkgName.map fun chr => if chr == '.' then '-' else chr
-  IO.FS.createDir dirName
+def validatePkgName (pkgName : String) : LogIO PUnit := do
+  if pkgName.isEmpty || pkgName.all (· == '.') || pkgName.any (· ∈ ['/', '\\']) then
+    error "illegal package name"
+  if pkgName.toLower ∈ ["init", "lean", "lake", "main"] then
+    error "reserved package name"
```

Comment:
I am tempted to say "some validation is better than none, but couldn't we restrict to `[0-9A-Za-z-_]*`?, but won't. :-)

Reply:
@semorrison My current reason for avoiding this is twofold. We might wish to permit non-English or Mathematical characters in package names for broader appeal. Second, this validation currently only applies to the `init`/`new` not to the package declaration itself, so any rules here can currently be circumvented. In the future, though, have a more restricted subset for package names in general would likely be a good idea.